activejob/lib/active_job/test_helper.rb

```diff
@@ -649,7 +649,10 @@ def flush_enqueued_jobs(only: nil, except: nil, queue: nil, at: nil)
 
       def prepare_args_for_assertion(args)
         args.dup.tap do |arguments|
-          arguments[:at] = round_time_arguments(arguments[:at]) if arguments[:at]
+          if arguments[:at]
+            at_range = arguments[:at] - 1..arguments[:at] + 1
```

Comment:
This is clearly wrong. `arguments[:at]` can be `Hash`, `Array` or even a proc. `arguments[:at] - ` will fail for those cases.

Reply:
Oh!  I did not realize it was allowed to be a proc when coming from `assert_enqueued_with` or `assert_performed_with`.  Their documentation indicates that only `arguments[:args]` can be a proc.  If that is the case, should I also modify their docs in a PR that fixes this?

Also, I'm having trouble seeing how a `Hash` or `Array` could work.  `arguments[:at]` seems to be matched using either [`arguments[:at].call(job[:at])`](https://github.com/jonathanhefner/rails/blob/fa2584fc2b11284c45d66a1764b455ed99c65d73/activejob/lib/active_job/test_helper.rb#L403) or [`arguments[:at] == job[:at]`](https://github.com/jonathanhefner/rails/blob/fa2584fc2b11284c45d66a1764b455ed99c65d73/activejob/lib/active_job/test_helper.rb#L405).  So `job[:at]` would have to be a `Hash` or `Array` for `arguments[:at]` to match in that case, no?

Reply:
I think in reality `arguments[:at]` are not `Hash` and `Array` my comment was based on `round_time_arguments` taking care of those cases. I don't we should support `Hash` and `Array` but we should support procs. We should update the documentation, yes.